apps/blockchain/lib/blockchain/block.ex

```diff
@@ -746,11 +801,29 @@ defmodule Blockchain.Block do
       |> add_rewards_to_block(db, chain.params[:block_reward])
 
     # The following checks Holistic Validity, as defined in Eq.(29)
-    errors = []
-      ++ if child_block.header.state_root == block.header.state_root, do: [], else: [:state_root_mismatch]
-      ++ if child_block.header.ommers_hash == block.header.ommers_hash, do: [], else: [:ommers_hash_mismatch]
-      ++ if child_block.header.transactions_root == block.header.transactions_root, do: [], else: [:transactions_root_mismatch]
-      ++ if child_block.header.receipts_root == block.header.receipts_root, do: [], else: [:receipts_root_mismatch]
+    errors =
```

Comment:
@germsvel that's definitely should be fixed before merging

Reply:
Very good catch. This is a disaster in the new format.

Reply:
Updated in 4b64ba226a8451bba26d94181176f4267c0383b0

Reply:
@germsvel it looks much better. good work. You also can add typespecs to new methods.

Reply:
Oh good catch. I'll add those and merge after that.

Reply:
Added typespecs in 5f22b330e6658f940df60d8a5f607cc88c2a5af4